src/openforms/translations/api/serializers.py

```diff
@@ -30,3 +36,118 @@ class LanguageInfoSerializer(serializers.Serializer):
+
+    def __init__(self, *args, nested_fields_mapping: Optional[dict] = None, **kwargs):
+        kwargs["source"] = "*"
+        self.nested_fields_mapping = nested_fields_mapping or {}
+        super().__init__(*args, **kwargs)
+
+    def get_parent_field(self, field_name):
+        """
+        Some fields may be nested in the parent serializer
+        """
+        if nested_path := self.nested_fields_mapping.get(field_name):
+            serializer = self.parent
+            for part in nested_path.split("."):
+                serializer = serializer.fields[part]
+            return serializer.fields[field_name]
+        return self.parent.fields[field_name]
+
+    def ignore_blank_errors(self, error: Union[Dict, List]):
+        if isinstance(error, dict):
+            for key, errors in error.items():
+                modified_errors = self.ignore_blank_errors(errors)
+                if modified_errors:
+                    error[key] = modified_errors
+                else:
+                    error.pop(key)
+            return error
+        else:
+            return [err for err in error if err.code != "blank"]
+
+    def run_validation(self, data):
+        """
+        Apply the original validation for the base fields to the translated fields
+        """
+        (is_empty_value, data) = self.validate_empty_values(data)
+        if is_empty_value:
+            return data
+
+        for language, fields in data.items():
+            self.parent.run_validators(fields)
+
+        return super().run_validation(data=data)
+
+    def get_fields(self):
+        return {code: serializers.JSONField() for code, label in settings.LANGUAGES}
+
+    def to_internal_value(self, data):
+        """
+        Run field validation and flatten the data, so it can be saved properly
+        """
+        ret = OrderedDict()
+        errors = OrderedDict()
+
+        fields = sorted(
+            get_translatable_fields_for_model(self.parent.get_model_class())
+        )
+        for language, _label in settings.LANGUAGES:
+            errors_for_lang = OrderedDict()
+            for field_name in fields:
+                parent_field = self.get_parent_field(field_name)
+                value = data.get(language, {}).get(field_name)
+                if value is None:
+                    if isinstance(parent_field, serializers.Serializer):
+                        value = parent_field.data
+                    else:
+                        value = ""
+
+                error = None
+                validate_method = getattr(parent_field, "validate_" + field_name, None)
+                try:
+                    validated_value = parent_field.run_validation(value)
+                    if validate_method is not None:
+                        validated_value = validate_method(validated_value)
+                except serializers.ValidationError as exc:
+                    error = self.ignore_blank_errors(exc.detail)
```

Comment:
I decided to ignore blank errors, because otherwise you'd always have to explicitly submit an empty string for `required` fields